web/src/components/roadmap/RoadmapSidebarList.tsx

```diff
@@ -18,13 +32,24 @@ export const RoadmapSidebarList = (props: Props): ReactElement => {
         })
     : [];
 
+  useEffect(() => {
+    if (
+      userData?.profileData.businessPersona === "FOREIGN" &&
+      userData?.preferences.visibleRoadmapSidebarCards.includes("graduation")
+    ) {
+      hideCard("graduation");
+    }
+  }, [userData, hideCard]);
+
   return (
     <>
+      <GraduationModal open={modalOpen} handleClose={() => setModalOpen(false)} />
+
       <h2>{Config.roadmapDefaults.sidebarHeading}</h2>
       <hr className="bg-roadmap-blue-dark hr-2px margin-bottom-4" />
 
       {visibleCardsOrderedByWeight.map((card: SidebarCardContent) => (
-        <RoadmapSidebarCard card={card} key={card.id} />
+        <RoadmapSidebarCard card={card} key={card.id} cTAOnClickMap={onlickMap} />
```

Comment:
does this need to passed in as a prop? could the card component own & look up the onClick handler in the map?